compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -103,6 +103,8 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
     val idSuffix =
       if (sym.isPrivate || allRefClasses.contains(sym.owner))
         sym.owner.ancestors.count(!_.isTraitOrInterface).toString
+      else if (sym.isOuterField)
+        sym.owner.name
```

Comment:
Woops. Sorry, I didn't realize my mind had switched languages ^^

@nicolasstucki This sounds good at first thought. I'll have to think more about this before taking a decision.